src/XML/src/FMI3/fmi3_xml_type.c

```diff
@@ -858,74 +951,217 @@ int fmi3_xml_handle_UInt8(fmi3_xml_parser_context_t* context, const char* data)
+    {
+        return NULL;
+    }
+    if (fmi3_xml_set_attr_uint64(context, elmID, fmi_attr_id_intervalCounter, 0 /*required*/,
+            &props->intervalCounter, fallbackProps->intervalCounter))
+    {
+        return NULL;
+    }
+    if (fmi3_xml_set_attr_uint64(context, elmID, fmi_attr_id_shiftCounter, 0 /*required*/,
+            &props->shiftCounter, fallbackProps->shiftCounter))
+    {
+        return NULL;
+    }
+    if (fmi3_xml_set_attr_float32(context, elmID, fmi_attr_id_intervalDecimal, 0 /*required*/,
+            &props->intervalDecimal, fallbackProps->intervalDecimal))
+    {
+        return NULL;
+    }
+    if (fmi3_xml_set_attr_float32(context, elmID, fmi_attr_id_shiftDecimal, 0 /*required*/,
+            &props->shiftDecimal, fallbackProps->shiftDecimal))
+    {
+        return NULL;
+    }
+    return props;
 }
 
 int fmi3_xml_handle_Clock(fmi3_xml_parser_context_t* context, const char* data) {
-    // TODO
-    assert(0);
-    return -1;
+    if (fmi3_xml_handle_SimpleType(context, data)) return -1;
+
+    if (!data) {
+        fmi3_xml_model_description_t* md = context->modelDescription;
+        fmi3_xml_variable_typedef_t* typeDef;
+        fmi3_xml_clock_type_props_t* props;
+        fmi3_xml_clock_type_props_t* defaultType = &context->modelDescription->typeDefinitions.defaultClockType;
+
+        props = fmi3_xml_parse_clock_type_properties(context, fmi3_xml_elmID_Clock, &defaultType->super);
+        if (!props) return -1;
+
+        // Get the typedef already created for SimpleType and set props
+        typeDef = jm_vector_get_last(jm_named_ptr)(&md->typeDefinitions.typeDefinitions).ptr;
+        typeDef->super.baseType = fmi3_base_type_clock;
+        typeDef->super.nextLayer = &props->super;
+    }
+    return 0;
 }
 
 int fmi3_xml_handle_String(fmi3_xml_parser_context_t* context, const char* data) {
+    if (fmi3_xml_handle_SimpleType(context, data)) return -1;
 
-    int res = fmi3_xml_handle_SimpleType(context, data);
-    if (res)
-        return res;
-
-    if(!data) {
+    if (!data) {
         fmi3_xml_model_description_t* md = context->modelDescription;
-        jm_named_ptr named;
-        fmi3_xml_variable_typedef_t* type;
+        fmi3_xml_variable_typedef_t* typeDef;
 
-        named = jm_vector_get_last(jm_named_ptr)(&context->modelDescription->typeDefinitions.typeDefinitions);
-        type = named.ptr;
-        type->super.baseType = fmi3_base_type_str;
-        type->super.nextLayer = &md->typeDefinitions.defaultStringType;
-    }
-    else {
-        /* don't do anything. might give out a warning if(data[0] != 0) */
-        return 0;
+        // Get the typedef already created for SimpleType and set props
+        typeDef = jm_vector_get_last(jm_named_ptr)(&context->modelDescription->typeDefinitions.typeDefinitions).ptr;
+        typeDef->super.baseType = fmi3_base_type_str;
+        typeDef->super.nextLayer = &md->typeDefinitions.defaultStringType;
     }
```

Comment:
Diff just looks strange. I changed _handle_SimpleType to be a one-liner and renamed type -> typeDef.